app/src/main/java/de/westnordost/streetcomplete/quests/opening_hours/ResurveyOpeningHours.kt

```diff
@@ -0,0 +1,82 @@
+package de.westnordost.streetcomplete.quests.opening_hours
+
+import android.util.Log
+import de.westnordost.osmapi.map.data.BoundingBox
+import de.westnordost.osmapi.map.data.Element
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.meta.OsmTaggings
+import de.westnordost.streetcomplete.data.osm.OsmElementQuestType
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.download.MapDataWithGeometryHandler
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao
+import de.westnordost.streetcomplete.data.osm.tql.getQuestPrintStatement
+import de.westnordost.streetcomplete.data.osm.tql.toGlobalOverpassBBox
+import de.westnordost.streetcomplete.quests.DateUtil
+
+class ResurveyOpeningHours(private val overpassServer: OverpassMapDataDao) : OsmElementQuestType<OpeningHoursAnswer> {
+    override val commitMessage = "resurvey opening hours"
+    override val icon = R.drawable.ic_quest_opening_hours
+    override fun getTitle(tags: Map<String, String>) = R.string.resurvey_opening_hours_title
+
+    override fun download(bbox: BoundingBox, handler: MapDataWithGeometryHandler): Boolean {
+        return overpassServer.getAndHandleQuota(getOverpassQuery(bbox)) { element, geometry ->
+            if (element.tags != null) {
+                // require opening hours that are supported
+                if (OpeningHoursTagParser.parse(element.tags["opening_hours"]!!) != null) {
+                    handler.handle(element, geometry)
+                } else {
+                    Log.wtf("AAAA", "opening_hours=" + element.tags["opening_hours"] + " was rejected as not representable in SC")
+                }
+            }
+
+        }
+    }
+
+    /** Note that 'newer:' query part will ensure that any edit,
+     * including adding or updating review marker like check_date or survey:date tags
+     * will cause OSM elements to become ineligible for this quest for reviewIntervalInDays days.
+     * It allows supporting check_date and any other survey markers without parsing of any tags.
+     * In addition, objects where tag was surveyed and confirmed to be correct by adding check_date:opening_hours
+     * are with larger delay, as it means this info on this specific object is relatively stable
+     */
+    private fun getOverpassQuery(bbox: BoundingBox): String {
+        val reviewIntervalInDays = 380
+        val date = "'${DateUtil.getOffsetDateString(-reviewIntervalInDays * 3)}T00:00:00Z'"
+        val dateTrippledTime = "'${DateUtil.getOffsetDateString(-reviewIntervalInDays * 3)}T00:00:00Z'"
```

Comment:
Copy-paste mistake?

Reply:
yes :(

Reply:
Also, better define two constants, one for openingHoursReviewInterval and one for elementReviewInterval (or something) than this triple-stuff.

Reply:
```
val resurveyCutoff = "'${DateUtil.getOffsetDateString(-reviewIntervalInDays)}T00:00:00Z'"
val repeatedResurveyCutoff = "'${DateUtil.getOffsetDateString(-reviewIntervalInDays * 3)}T00:00:00Z'"
```